pkg/reconciler/common/stages.go

```diff
@@ -83,7 +83,9 @@ type ManifestFetcher func(ctx context.Context, instance v1alpha1.KComponent) (*m
 // manifest is captured in a closure before any stage might mutate the
 // instance status, e.g. Install.
 func DeleteObsoleteResources(ctx context.Context, instance v1alpha1.KComponent, fetch ManifestFetcher) Stage {
-	if TargetVersion(instance) == instance.GetStatus().GetVersion() {
+	version := TargetVersion(instance)
+	if version == instance.GetStatus().GetVersion() && targetManifestPath(version,
+		instance) == installedManifestPath(version, instance) {
```

Comment:
Maybe use a more idiomatic style of line wrapping here?
```suggestion
	if version == instance.GetStatus().GetVersion() &&
		targetManifestPath(version, instance) == installedManifestPath(version, instance) {

```